app-creator/client/src/widgets/templates-tab/templates-tab.ts

```diff
@@ -134,13 +162,57 @@ export class TemplatesTab extends connect(store)(LitElement) {
     });
   }
 
-  createSelectionCallback(template: string): VoidFunction {
+  /**
+   * Callback triggered on template card selection.
+   */
+  handleTemplateCardSelection(id: string) {
     return () => {
-      store.dispatch(setSelectedTemplate(JSON.parse(template)));
-      this.requestUpdate();
+      this.requestedTemplateID = id;
+      if (this.templateChangeConfirmationDialog) {
+        this.templateChangeConfirmationDialog.open();
+      }
     };
   }
 
+  /**
+   * Callback triggered on template change confirmation.
+   */
+  handleTemplateChangeConfirmation() {
+    const template = templatesManager
+      .getTemplates()
+      .find(({ id }) => id === this.requestedTemplateID)?.template;
+
+    if (template) {
+      try {
+        storeTemplateInLocalStorage();
+
+        // Replace the redux store with the new template and trigger a re-render.
+        const templateJSON = JSON.parse(template);
+        store.dispatch(setSelectedTemplate(templateJSON));
+
+        transferData();
+      } catch (e) {
+        console.error(e);
+      }
+    }
+
+    if (this.templateChangeConfirmationDialog) {
+      this.templateChangeConfirmationDialog.close();
+    }
+
+    this.requestUpdate();
+  }
+
+  /**
+   * Callback triggered on template change dismiss.
+   */
+  handleTemplateChangeDismiss() {
+    this.requestedTemplateID = '';
+    if (this.templateChangeConfirmationDialog) {
```

Comment:
Since the non-null assertion is placed after templateChangeConfirmationDialog in the @query annotation, we could probably safely assume that this.templateChangeConfirmationDialog exists?